doc/lsst.daf.butler_migrate/concepts.rst

```diff
@@ -0,0 +1,142 @@
+    1a93ca89bc27 -> b4073b638219 (collections, collections-SynthIntKeyCollectionManager), Migration script for SynthIntKeyCollectionManager 0.3.0.
+    079a1bc77f25 -> 1a93ca89bc27 (collections, collections-SynthIntKeyCollectionManager), Migration script for SynthIntKeyCollectionManager 0.2.0.
+    8d2e9de2f21f -> 079a1bc77f25 (collections, collections-SynthIntKeyCollectionManager), Migration script for SynthIntKeyCollectionManager 0.1.0.
+    5e75e5391603 -> 93341d68b814 (collections-NameKeyCollectionManager, collections) (head), Migration script for NameKeyCollectionManager 2.0.0.
+    e866114194e1 -> 5e75e5391603 (collections-NameKeyCollectionManager, collections), Migration script for NameKeyCollectionManager 1.0.0.
+    3ce2d3adf1f5 -> e866114194e1 (collections-NameKeyCollectionManager, collections), Migration script for NameKeyCollectionManager 0.3.0.
+    a56af31bd899 -> 3ce2d3adf1f5 (collections-NameKeyCollectionManager, collections), Migration script for NameKeyCollectionManager 0.2.0.
+    8d2e9de2f21f -> a56af31bd899 (collections-NameKeyCollectionManager, collections), Migration script for NameKeyCollectionManager 0.1.0.
+    <base> -> 8d2e9de2f21f (collections) (branchpoint), This is an initial pseudo-revision of the 'collections' tree.
+
+In this listing ``<base>`` meas a special initial Alembic pseudo-revision, essentially an empty schema.
+Revision ``8d2e9de2f21f`` is a root of the whole tree for ``collections`` manager type, and each class' branch originates from that root revision.
+The ``(head)`` notifies the latest revision in a branch.
+
+It must be noted that while logically these revisions and branches may look similar to other revision systems (e.g. ``git``), there are also significant differences in their behavior.
+One example is that revisions "inherit" the names of the branches, this is why every revision in the above list has ``collections`` branch name associated with it, and one of the class branch names.
+Another big difference is in howe Alembic handles branch merges, which requires use of special one-shot migrations for switching between implementation classes/branches (described below).
+
+
+Migration scripts
+=================
+
+Alembic revision history is defined as a collection of migration scripts, which are Python scripts with predefined structure.
+Every script represents a single step in migration tree and defines several attributes, two most important among them are its target revision and a parent (or "down") revision.
+Alembic does not define a separate structure for describing migration trees, that structure is deduced from the whole collection of migration scripts by connecting their parent revisions with other scripts.
+``daf_butler_migrate`` keeps the collection of migration scripts in ``migrations`` folder, with a separate sub-folder for each manager type.
```

Comment:
These files are used by Alembic, and I'm not sure that it can find them in pkg_resources. I need to learn more about how Alembic works and see if we can trick it to do what we want, but I'll leave it to a later ticket.